action.yml

```diff
@@ -0,0 +1,58 @@
+name: 'Validate Product Version'
+description: 'Check that product version output includes expected version string'
+inputs:
+  expected_version:
+    type: string
+    description: 'Expected version string'
+    required: true
+  artifact_path:
+    type: string
+    description: "Path to artifacts to test"
+    required: true
+  soft_fail:
+    type: string
+    # this input should only be set by CRT workflows or internal tests
+    description: 'If non-empty, lack of support is not a fatal error'
+
+runs:
+  using: "composite"
+  steps:
+
+    - name: Check Product Support
+      id: check-product-support
+      shell: bash
+      run: |
+        tool="${{ inputs.artifact_path }}"
+        tool="${tool##*/}"
+        supported="$(${{ github.action_path }}/check-product-support "${{ inputs.artifact_path }}")"
+        echo "supported=$supported" > "$GITHUB_OUTPUT"
+        echo "::debug::output: $supported"
+
+        if [ "$supported" != 'true' ] ; then
+          echo "::error ${tool} is not supported by this action." 1>&2
+          [ -n "${{ inputs.soft_fail }}" ] && exit 0 # exit 1 if soft_fail unset
```

Comment:
Instead of aborting (that is what's happening here right?) the test on unsupported products, is it feasible perhaps to take a more TDD approach and instead to test for failure? That way when a product starts succeeding we report the unexpected change so are hardcoded list gets updated.

I dunno, just a thought. Kinda depends on how many unsupported products we think we will catch.

Reply:
Reporting how?  I don't want to try to pull slack notification (or secrets) into this, so anything that doesn't fail the workflow won't get seen -- the output is collapsed, the workflow gets a green check, and the release process continues.

We don't actually need the hardcoded list to be updated, as product teams that want this can call the action directly from their build.yml workflow.  They can opt-in with no action on our side.